src/Entity/Type/RefundEnumType.php

```diff
@@ -30,31 +31,37 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
         return 'VARCHAR(256)';
     }
 
-    public function convertToPHPValue($value, AbstractPlatform $platform): RefundType
+    public function convertToPHPValue($value, AbstractPlatform $platform): RefundTypeInterface
     {
-        if (!RefundType::isValid($value)) {
+        if ($value instanceof RefundType && !$value::isValid($value)) {
```

Comment:
We should probably check here `RefundTypeInterface` instead of `RefundType`

Reply:
Yes but interface doesn't have `isValid`, and is problem with code analyse. I can check `$value instanceof RefundTypeInterface && $value instanceof Enum` but checking with `RefundType` work because we need overwrite this class when we create new `RefundType` and `instanceof` see this object as old `RefundType` because is his parent